pkg/check/feed/feed.go

```diff
@@ -0,0 +1,174 @@
+package feed
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/bee/v2/pkg/crypto"
+	"github.com/ethersphere/bee/v2/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/bee/api"
+	"github.com/ethersphere/beekeeper/pkg/beekeeper"
+	"github.com/ethersphere/beekeeper/pkg/logging"
+	"github.com/ethersphere/beekeeper/pkg/orchestration"
+)
+
+// Options represents check options
+type Options struct {
+	PostageAmount int64
+	PostageDepth  uint64
+	PostageLabel  string
+	NUpdates      int
+	RootRef       string
+}
+
+// NewDefaultOptions returns new default options
+func NewDefaultOptions() Options {
+	return Options{
+		PostageAmount: 1000,
+		PostageDepth:  17,
+		PostageLabel:  "test-label",
+		NUpdates:      2,
+	}
+}
+
+// compile check whether Check implements interface
+var _ beekeeper.Action = (*Check)(nil)
+
+// Check instance.
+type Check struct {
+	logger logging.Logger
+}
+
+// NewCheck returns a new check instance.
+func NewCheck(logger logging.Logger) beekeeper.Action {
+	return &Check{
+		logger: logger,
+	}
+}
+
+func (c *Check) Run(ctx context.Context, cluster orchestration.Cluster, opts interface{}) (err error) {
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	if o.RootRef != "" {
+		c.logger.Infof("running availability check")
+		return c.availability(ctx, cluster, o)
+	}
+	return c.regular(ctx, cluster, o)
+}
+
+func (c *Check) availability(ctx context.Context, cluster orchestration.Cluster, o Options) error {
+	ref, err := swarm.ParseHexAddress(o.RootRef)
+	if err != nil {
+		return fmt.Errorf("invalid root ref: %w", err)
+	}
+
+	nodeNames := cluster.FullNodeNames()
+	nodeName := nodeNames[0]
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+
+	client := clients[nodeName]
+	_, _, err = client.DownloadFile(ctx, ref, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Check) regular(ctx context.Context, cluster orchestration.Cluster, o Options) error {
```

Comment:
maybe rename to something like fullFeedsCheck with some comments above the function?